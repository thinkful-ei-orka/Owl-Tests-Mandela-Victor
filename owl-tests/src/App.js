import React from 'react';
import participantStore from './participantstore';
import participantList from './participantlist';
// import Participants from './participant'

import './App.css'

function App() {

  const participants = participantStore.map(participant => (
    <Participant key={participantStore.participant[id]} name={participantStore.participant[name]} avatar={participantStore.participant[avatar]} inSession={participantStore.participant[inSession]} onStage={participantStore.participant[onStage]} />
));

  // const participants = participantStore.participants;
  return (
    <main className='App'>
      <div className='App-Participants'>
          {participants}
      </div>
      {/* <div className='App-ChatLog'>
          {chatlog}
      </div>
      <div className='App-Stage'>
          {stage}
      </div> */}
    </main>
  );
}

export default App;