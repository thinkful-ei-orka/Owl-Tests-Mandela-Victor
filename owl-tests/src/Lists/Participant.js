import React from 'react';
import participants from '../store'

// import './participants.css'

export default function Participant(props) {
    let stageStatus = '' ;
    // props.onStage = true ? stageStatus = 'on stage' : stageStatus = '';
    // props.inSession = true ? sessionStatus = 'in session' : sessionStatus = '';

         return (
             <div className='Participant' id={props.id}>
                 <h3 className='Participant-name'>{props.name}</h3>
                 <img className='Participant-avatar' src={props.avatar} />
                 <p className='Participant-inSession'>{props.inSession}</p>
                 <p className='Participant-onStage'>{stageStatus}</p>
             </div>
         );
     }