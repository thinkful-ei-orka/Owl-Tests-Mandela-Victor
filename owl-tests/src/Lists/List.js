import React from 'react';
import store from '../store.js';
import './List.css'
import Participant from './Participant'

export default function List(props) {
    
    const participants = store.participants.map(participant => (
         <Participant 
         key={participant.id} 
         name={participant.name} 
         avatar={participant.avatar} 
         inSession={participant.inSession} 
         onStage={participant.onStage} 
         />
     ));
 
     return (
         <section className='List'>
             <ul className='List-Participants'>
                 {participants}
             </ul>
         </section>
     );
 }