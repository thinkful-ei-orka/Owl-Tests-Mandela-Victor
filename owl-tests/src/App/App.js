import React from 'react';
import participants from '../participantStore.js';
import List from '../Lists/List.js';
import Stage from '../Stage/Stage.js'
import Chat from '../Chat/Chat.js'

import './App.css'

function App(props) {
  //will props.participants work?
  return (
    <main className='App'>
      <List participants={props.participants} />
      <Chat participants={props.participants} chatEvents={props.chatEvents}/>
      <Stage participants={props.participants} />
    </main>
  );
}

export default App;