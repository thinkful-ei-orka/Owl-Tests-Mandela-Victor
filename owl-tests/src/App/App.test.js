import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../App/App';
import '../Lists/List'

it('renders', () => {
     const div = document.createElement('div');
     ReactDOM.render(<App />, div);
     ReactDOM.unmountComponentAtNode(div);
})

it('renders', () => {
     const tree = renderer
     .create(<App className='App'/>)
     .toJSON();
     expect(tree).toMatchSnapshot();
})