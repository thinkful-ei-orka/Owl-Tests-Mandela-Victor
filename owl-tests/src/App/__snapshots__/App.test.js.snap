// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<main
  className="App"
>
  <section
    className="List"
  >
    <ul
      className="List-Participants"
    >
      <div
        className="Participant"
      >
        <h3
          className="Participant-name"
        >
          Koren Templeton
        </h3>
        <img
          className="Participant-avatar"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
        <p
          className="Participant-inSession"
        />
        <p
          className="Participant-onStage"
        >
          
        </p>
      </div>
      <div
        className="Participant"
      >
        <h3
          className="Participant-name"
        >
          Caty Flucker
        </h3>
        <img
          className="Participant-avatar"
          src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
        />
        <p
          className="Participant-inSession"
        />
        <p
          className="Participant-onStage"
        >
          
        </p>
      </div>
      <div
        className="Participant"
      >
        <h3
          className="Participant-name"
        >
          Axe Kubicka
        </h3>
        <img
          className="Participant-avatar"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
        <p
          className="Participant-inSession"
        />
        <p
          className="Participant-onStage"
        >
          
        </p>
      </div>
      <div
        className="Participant"
      >
        <h3
          className="Participant-name"
        >
          Frank Runciman
        </h3>
        <img
          className="Participant-avatar"
          src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
        />
        <p
          className="Participant-inSession"
        />
        <p
          className="Participant-onStage"
        >
          
        </p>
      </div>
      <div
        className="Participant"
      >
        <h3
          className="Participant-name"
        >
          Ashla Attwool
        </h3>
        <img
          className="Participant-avatar"
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
        <p
          className="Participant-inSession"
        />
        <p
          className="Participant-onStage"
        >
          
        </p>
      </div>
    </ul>
  </section>
  <section
    className="Chat"
  >
    <ul
      className="Chat-list"
    >
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Axe Kubicka
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Caty Flucker
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Frank Runciman
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Koren Templeton
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Koren Templeton
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Koren Templeton
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Ashla Attwool
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Axe Kubicka
          </span>
          null
        </h3>
      </li>
      <li
        className="emoji-message"
      >
        <h3>
          <span>
            Axe Kubicka
          </span>
          null
        </h3>
      </li>
    </ul>
  </section>
  <div
    className="Stage"
  >
    <div
      className="Stage-Participants"
    >
      <div
        className="Stage-Participant"
        id={1}
      >
        <h3>
          Koren Templeton
        </h3>
        <img
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
      </div>
      <div
        className="Stage-Participant"
        id={5}
      >
        <h3>
          Ashla Attwool
        </h3>
        <img
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
  </div>
</main>
`;
